### PR TITLE
Improvements to the release process.

### PR DESCRIPTION
**Pull request by bac**

### CODE REVIEW

**Comment by bac** [comment]
QA:

If you want to QA the new makefile targets without screwing everything up, try this:

```
# Use staging.
export JUJU_CHARMSTORE=https://api.staging.jujucharms.com/charmstore
YELLOW_CS_URL="cs:~bac/trusty/juju-gui" make publish-yellow
```

You'll need to use your own id above.  Quotes necessary to prevent ~name expansion if your local userid and the one on jujucharms.com are the same.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/80/
Test PASSed.

**Comment by jrwren** [comment]
:+1: nice to have this level of integration in deployment.

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -88,36 +88,21 @@ release.
...
 
     charm2 info --include=id,perm cs:~yellow/juju-gui
-    charm2 info --include=id,perm cs:~yellow/development/juju-gui
 
 Next, to upload the charm, go to the charm source directory and do:
 
-    make clean-tests
-    # NB: trusty is the default series unless specified in metadata.yaml.
-    charm2 upload . cs:~yellow/trusty/juju-gui
+    make publish-yellow
```
Comment (jaycee) [inline RELEASE_PROCESS.md]: So, I see both publish commands have `clean` as a req, but here we're suggesting `clean-tests`; is that just an out of date target or should publish commands use that ad the req?

**Comment by jaycee** [comment]
:+1: with one possible point noted. I will QA shortly.

File: RELEASE_PROCESS.md (hunk 1 above)
Comment (bac) [inline RELEASE_PROCESS.md]: the `make clean-tests` is removed. it was incorrect.  clean is better because it removed the .pyc files.

**Comment by bac** [comment]
:shipit:

File: RELEASE_PROCESS.md (hunk 1 above)
Comment (jaycee) [inline RELEASE_PROCESS.md]: Cool, thanks.

**Comment by jaycee** [comment]
QA :ok:

**Comment by hatched** [comment]
:+1: Thanks bac, looks good!

**Comment by bac** [comment]
:shipit:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/87/
Test PASSed.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-charm-merge

**Pull request merged**